db-client/src/main/java/com/splicemachine/db/client/net/NetConnection.java

```diff
@@ -2162,5 +2167,17 @@ protected void writeXATransactionStart(Statement statement) throws SqlException
     public void setServerPrincipal(String serverPrincipal) {
         this.serverPrincipal = serverPrincipal;
     }
+
+    @SuppressFBWarnings(value = "DM_DEFAULT_ENCODING", justification = "DB-11046")
+    private void flowUSRIDTokenconnect(String authenticator, String token) throws SqlException {
```

Comment:
Isn't the "user id" embedded into the token? Why do we specify it separately too?

Reply:
yes, it's embedded. It was done to simplify an implementation. If a user id not provided in parameters, it's filled as 'splice' by default and I can change it only after parsing of token, but as I remember there was a problem, that some parts were already initialized with 'splice' as user id. I have to look again to improve it.

Reply:
userId is taken from token now